code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -607,6 +609,24 @@
 	else
 		footstep++
 
+/obj/item/clothing/suit/space/hardsuit/ancient/equipped(mob/user, slot)
+	. = ..()
+	if (slot == slot_wear_suit)
+		if (mobhook && mobhook.parent != user)
+			QDEL_NULL(mobhook)
+		if (!mobhook)
+			mobhook = user.AddComponent(/datum/component/redirect, list(COMSIG_MOVABLE_MOVED), CALLBACK(src, .proc/on_mob_move))
+	else
+		QDEL_NULL(mobhook)
+
+/obj/item/clothing/suit/space/hardsuit/ancient/dropped()
+	. = ..()
+	QDEL_NULL(mobhook)
+
+/obj/item/clothing/suit/space/hardsuit/ancient/Destroy()
+	QDEL_NULL(mobhook) // mobhook is not our component
```

Comment:
What do you mean by that comment ?

Reply:
The mobhook is not a component of the instance of `/obj/item/clothing/suit/space/hardsuit/ancient`, but rather, if it exists, a component of the `/mob` wearing the suit. Therefore, when the suit is deleted, the mobhook component on the mob must be deleted as well, otherwise the callback will runtime every time.